logstash-core/spec/logstash/webserver_spec.rb

```diff
@@ -21,6 +21,8 @@
 require "spec_helper"
 require "open-uri"
 require "webmock/rspec"
+require "faraday"
```

Comment:
Out of curiosity, do you know what changed that forced this change? Is it part of this change? Or something outside of it?

Reply:
This is caused by the removal of [elasticsearch_client.rb](https://github.com/elastic/logstash/pull/16794/commits/b47188abdf1c626c1b132c13d536635180f4d805#diff-df536490516d7a5ffd225497283c381778be1718b2d7b9b285721c65426fc027L18-L19) and the import in [runner_spec.rb](https://github.com/elastic/logstash/pull/16794/commits/b47188abdf1c626c1b132c13d536635180f4d805#diff-a4e3e6b841aa2183c5550359b0732036b0f8db6f1690434058be301416a49908L25). The elasticsearch related classes somehow include Faraday.

The import in other spec files affect the global namespace for the entire test suite. You can add `require "faraday"` in runner_spec and get a green run of `./gradlew :logstash-core:rubyTests`, but it will fail when running a single rspec file as webserver_spec itself doesn't link faraday.